File: code_challenges/02_reverseString.js
/*
Expected function: `reverseString`
Write a function called *reverseString* that accepts a string and returns a reversed copy of the string.

Input Example:

```
'hello world'
'asdf'
'CS rocks!'
```

Output Example:

```
'dlrow olleh'
'fdsa'
'!skcor SC'
```

*/ 

// Using a for loop:
function reverseString(str) {
  let reversedString = ''
  for(let i = 0; i < str.length; i++) {
    reversedString = str[i] + reversedString
  }
  return reversedString
}

console.log(
  reverseString('hello world')
)

File: code_challenges/clojure_solutions/02-reverse-string.clj
;; Expected function: `reverseString`
;; Write a function called *reverseString* that accepts a string and returns a reversed copy of the string.
;; 
;; Input Example:
;; 
;; ```
;; 'hello world'
;; 'asdf'
;; 'CS rocks!'
;; ```
;; 
;; Output Example:
;; 
;; ```
;; 'dlrow olleh'
;; 'fdsa'
;; '!skcor SC'
;; ```

(ns lambda-school
  (:require [clojure.string :as str]))

(defn reverse-string [s]
  (apply str (reduce conj () (seq s))))

(reverse-string "hello") ;; => "olleh"
